### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by vbergmann**

### CODE REVIEW

File: src/main/java/org/aludratest/service/smtp/SmtpService.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	@ConfigProperty(name = "password", description = "The password name to use when connecting to a mail server.", 
+			type = int.class, required = false),
+	@ConfigProperty(name = "auth", description = "If true, attempt to authenticate the user using the " +
+			"AUTH command. Defaults to false.", 
+			type = boolean.class, required = false, defaultValue = "false"),
+	@ConfigProperty(name = "starttlsEnable", description = "enables the use of the STARTTLS command (if " +
+			"supported by the server) to switch the connection to a TLS-protected connection before issuing " +
+			"any login commands. Defaults to false.", 
+			type = boolean.class, required = false, defaultValue = "false"),
+	@ConfigProperty(name = "sslCheckserveridentity", description = "If set to true, checks the server identity " +
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/service/smtp/SmtpService.java]: I would prefer full camel case: sslCheckServerIdentity

File: src/main/java/org/aludratest/service/smtp/impl/SmtpAction.java [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			sendMailWithAttachments(sender, recipients, subject, text, attachmentUris, mimeTypes);
+		} else {
+			sendPlainTextMail(sender, recipients, subject, text);
+		}
+	}
+
+
+	// private helpers ---------------------------------------------------------
+	
+	private void sendPlainTextMail(String sender, String[] recipients, String subject, @AttachParameter("Message text") String text) {
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/service/smtp/impl/SmtpAction.java]: Are you sure `@AttachParameter` will work here (is required here)?

File: src/main/java/org/aludratest/service/smtp/impl/SmtpAction.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+			Message message = createMessage(sender, recipients, subject);
+			message.setText(text);
+			this.recentEmail = new EmailData(sender, recipients, subject, text);
+			saveChangesAndSendMessage(message);
+		} catch (MessagingException e) {
+			throw new TechnicalException("Sending email failed", e);
+		}
+	}
+
+	private void sendMailWithAttachments(String sender, String[] recipients, String subject, @AttachParameter("Message text") String text,
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/service/smtp/impl/SmtpAction.java]: Same as above

File: src/main/java/org/aludratest/service/smtp/SmtpInteraction.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * @author Volker Bergmann
+ */
+public interface SmtpInteraction extends Interaction {
+
+	public void sendMail(
+			String sender, 
+			String[] recipients, 
+			String subject, 
+			@AttachParameter("Message text") String text,
+			String[] filesToAttach,
```
Comment (falbrech-hsdg) [inline src/main/java/org/aludratest/service/smtp/SmtpInteraction.java:41]: Javadoc required: What format shall the String for each file have? Additionally, would be great if they were also attached to the log.

**Review by falbrech-hsdg — CHANGES_REQUESTED** [review]
Please check comments. Also, a test case should be added which checks that the message text is really attached to the test log. See LoggingFileServiceTest for an example.

File: src/main/java/org/aludratest/service/smtp/SmtpInteraction.java (hunk 4 above)
Comment (vbergmann) [reply src/main/java/org/aludratest/service/smtp/SmtpInteraction.java:41]: Good review! I provided the requested changes and added a TODO note that currently the email attachments are not logged.

**Review by falbrech-hsdg — APPROVED** [review]

**Pull request merged**